.github/workflows/nix.yml

```
@@ -0,0 +1,27 @@
+name: Check Nix Formatting/Build
+
+on:
+  push:
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  nix-fmt:
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Nix
+        uses: cachix/install-nix-action@v31
+      - name: Check formatting
+        run: nix fmt -- --fail-on-change
+  nix-build:
+    name: Nix Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Nix
+        uses: cachix/install-nix-action@v31
+      - name: Build with nix
+        run: nix build
```

using nix flake check would make more sense here
```suggestion
  nix-flake-check:
    name: Nix Flake Check
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - name: Install Nix
        uses: cachix/install-nix-action@v31
      - name: Build with nix
        run: nix flake check
```

---

Isn't `nix flake check --all-systems` better?
Btw thank you for all your reviews! I'm really learning quite a lot thanks to them :)

---

all systems would make sense, also keeping the build in addition to the check would be a good idea.